### PR TITLE
Reset zoom feature added (fixes #3848)

### PR DESCRIPTION
**Pull request by rohit645**
This pull request makes the following changes:
- UI enhancement.
- Adds the reset zoom button. As user clicks on the reset zoom button map is set to default center [latitude, longitude] and zoom level.

Testing checklist:
- [x] Verified code style using jscs
- [x] Verified eslint checklist 
- [x] I certify that I ran my checklist

Fixes ushahidi/platform#3848 .

**Demo**
resetZoom

### CODE REVIEW

**Comment by rowasc** [comment]
Wow this looks fantastic, @rohit645 . Thank you so much! 
I'm assigning @Angamanga for a detailed review and she'll get back to you ASAP.

File: app/common/services/maps.js [hunk 1]
```diff
@@ -58,11 +59,34 @@ function Maps(ConfigEndpoint, L, _, CONST) {
...
+                id: 'reset-button',
+                position: 'bottomleft',
+                type: 'replace',
+                leafletClasses: true,
+                states:[{
+                    // specify different icons and responses for your button
+                    stateName: 'reset-button',
+                    onClick: function() {
+                        var defaultview = defaultValues();
+                        map.setView([defaultview.lat, defaultview.lon], defaultview.zoom);
```
Comment (Angamanga) [inline app/common/services/maps.js]: Send config to the defaultValues-function

File: app/common/services/maps.js [hunk 2]
```diff
@@ -140,4 +163,14 @@ function Maps(ConfigEndpoint, L, _, CONST) {
             popupAnchor: [0, 0 - size[1]]
         });
     }
+
+    // default view is centered on nairobi
+    function defaultValues(config) {
```
Comment (Angamanga) [inline app/common/services/maps.js]: Here, if you use config.center and config.zoom, you will get the default-values for the deployment, this defaults to Nairobi if the user has not set any own values but sometimes they have (this is set in Settings->General).

File: app/common/services/maps.js [hunk 3]
```diff
@@ -58,11 +59,34 @@ function Maps(ConfigEndpoint, L, _, CONST) {
...
+                }]
+            });
+
+            resetButton.addTo(map);
 
             return map;
         });
     }
 
+    function setDefaultView() {
```
Comment (Angamanga) [inline app/common/services/maps.js]: Is this function used? I cannot find it but maybe its a Leaflet-default-thing?

File: app/common/services/maps.js [hunk 4]
```diff
@@ -112,9 +136,9 @@ function Maps(ConfigEndpoint, L, _, CONST) {
 
     function defaultConfig() {
         return {
-            scrollWheelZoom: false,
+            scrollWheelZoom: true,
             center: [-1.2833, 36.8167], // Default to centered on Nairobi
-            zoom: 8,
+            zoom: 3,
```
Comment (Angamanga) [inline app/common/services/maps.js]: Nice catch! Zoom 8 is really too much :)

File: sass/overrides/_leaflet.scss [hunk 5]
```diff
@@ -65,3 +65,10 @@ input.leaflet-control-layers-selector {
 .leaflet-map-pane svg.iconic {
     position: initial;
 }
+
+//  Override styles for reset button
+#reset-button {
+    color: black;
+    background-color: white;
+    padding: 8px 8px;
+}
```
Comment (Angamanga) [inline sass/overrides/_leaflet.scss]: Could you check the styles for the small maps in edit-mode and general-view? There is an odd shadow-outline outside of the icon: 
Screenshot 2020-02-24 at 10 46 01

File: app/common/services/maps.js (hunk 3 above)
Comment (rohit645) [reply app/common/services/maps.js]: Yes, you're right :). I should have seen that earlier. This piece of code is redundant.

> Is this function used? I cannot find it but maybe its a Leaflet-default-thing?

Yes, you are right :). This piece of code is redundant. I should have seen that earlier.

Comment (rohit645) [reply app/common/services/maps.js]: > Is this function used? I cannot find it but maybe its a Leaflet-default-thing?

Yes, you are right. This piece of code is redundant. I should have seen that earlier :)

**Comment by rohit645** [comment]
Thanks for the feedback @Angamanga . I have made the required changes. Kindly review and let me know if any further changes are required.

File: app/index.html [hunk 6]
```diff
@@ -14,6 +14,7 @@
         <meta name="apple-itunes-app" content="app-id={{appStoreId}}, app-argument={{currentFullUrl}}>">
 
         <link rel="manifest" href="/manifest.json">
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (Angamanga) [inline app/index.html]: Remove this and use an icon from the pattern-library instead.

File: sass/overrides/_leaflet.scss (hunk 5 above)
Comment (Angamanga) [reply sass/overrides/_leaflet.scss]: Great work getting rid of this :)

File: app/main/posts/views/post-view-map.html [hunk 7]
```diff
@@ -3,7 +3,7 @@
     <div ui-view="mode-context" noanimation="true" filters="filters"></div>
     <main role="main">
 		<post-toolbar is-loading="isLoading" filters="filters" edit-mode="{ editing: false }"></post-toolbar>
-    	<div class="map-view">
+    	<div class="map-view" id="map-view">
 		    <div id="map-full-size" class="map full-size" ng-show="!noui"></div>
```
Comment (Angamanga) [inline app/main/posts/views/post-view-map.html]: I don't think the id is needed here, we use the ids in the child-divs instead, one for using the normal view you see when navigating to the map-view and one for specific use cases when no ui around the map is needed.

**Comment by rohit645** [comment]
Hey @Angamanga , I have made the suggested changes. Please review my PR.

**Comment by Angamanga** [comment]
@Angamanga Note to self, merge after release

**Pull request merged**